src/seedu/addressbook/data/person/Address.java

```diff
@@ -10,23 +10,56 @@
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_INVALID = "Person address must be in the following format a/BLOCK, STREET, UNIT, POSTAL_CODE ";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
+    public static final int BLOCK_NUMBER = 0;
+    public static final int STREET_NUMBER = 1;
+    public static final int UNIT_NUMBER = 2;
+    public static final int POSTALCODE_NUMBER = 3;
+    
+    public static final int REQUIRED_NUMBER_OF_ADDRESS_FIELDS = 4;
+    
     public final String value;
+    
+    
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final PostalCode postalCode;
+    
     private boolean isPrivate;
 
     /**
      * Validates given address.
-     *
+     * 
+     * The address must be entered in the following format a/BLOCK, STREET, UNIT, POSTAL_CODE 
+     * e.g. a/123, Clementi Ave 3, #12-34, 231534
+     * 
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+
         this.value = address;
+        
+        final String[] splitAddress = address.split(", ");
+       
+        if (splitAddress.length < REQUIRED_NUMBER_OF_ADDRESS_FIELDS) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_INVALID);
+        }
+        
+        this.block = new Block(splitAddress[BLOCK_NUMBER]);
+        this.street = new Street(splitAddress[STREET_NUMBER]);
+        this.unit = new Unit(splitAddress[UNIT_NUMBER]);
+        this.postalCode = new PostalCode(splitAddress[POSTALCODE_NUMBER]);
+        
     }
+    
+    
```

Comment:
Add a compareTo function to all the address field classes. First the name of the field will be compared through the use of strings (I.e. "Block"), and then if the type matches, the content of the field will then be compared.

I will add the function that returns this
`block.getBlock() + ", " + street.getStreet() + ", " + unit.getUnit() + ", " + postalCode.getPostalCode()`